### ISSUE TITLE
Cannot access AutoIncrement property

### ISSUE THREAD

**Issue by vertisan** [nelmio/alice#1045]
Hi,  
I have some entities:  
```php
class FirstEntity
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="SEQUENCE")
     * @ORM\Column(name="id", type="integer")
     */
    protected int $id;

    public function getId(): int
    {
        return $this->id;
    }

    // ....
}

class SecondEntity
{
    protected int $firstEntityId;

    public function __construct(FirstEntity $firstEntity)
    {
        $this->firstEntityId = $firstEntity->getId();
    }
    // ....
}
```
and fixtures for it:  
```yaml
App\Entity\FirstEntity:
  first_{1..10}:

App\Entity\SecondEntity:
  second_{1..10}:
    __construct:
      - '@first_<current()>'
```

When I tried to load these fixtures, I'm getting error:  
```
Typed property App\Entity\FirstEntity::$id must not be accessed before initialization
```
so it means, `$id` of `FirstEntity` is not filled by value from database.  
How objects from fixtures are loaded and how to resolve this problem?

**Comment by theofidry** [comment]
>so it means, $id of FirstEntity is not filled by value from database.

Indeed: alice only takes care of creating the objects, not persisting them. And when they do, like done in https://github.com/theofidry/AliceDataFixtures, they tend to persist and flush all the objects in one go. Otherwise you run into the problem of knowing what object can be added to the DB first and it's a very complicated problem.

>How objects from fixtures are loaded and how to resolve this problem?

You have 2 ways to do so:

- Have some complete objects beforehand that you can pass to the loader and they will be accessible from the fixtures. For example if you pass `['foo' => new stdClass()]` to the loader, you can refer to `@foo` within the fixtures.
- Not do that*

*: What you are doing is not considered a good practice. One one side you have `FirstEntity` that is not a complete object until persisted (due to its auto-generated ID) and on the other you are not setting `SecondEntity` <-> `FirstEntity` relation with objects but with IDs. It's really not a good mix in general and it's highly recommended to either have:

- `SecondEntity` <-> `FirstEntity` relation with objects i.e. you have `SecondEntity#firstEntity` instead of `SecondEntity#firstEntityId` (in other words you don't need to access to `FirstEntity` potentially incomplete state)
- Have a relation with IDs, i.e. you rely on object "reference" (in which case I would say you should pass directly `FirstEntityId` instead of the `FirstEntity` instance) with pre-assigned IDs. I.e. you don't need to wait on the object to be persisted from the DB to be usable as you wish
